packages/svelte-vscode/src/typescript/findFileReferences.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    commands,
+    ExtensionContext,
+    ProgressLocation,
+    Uri,
+    window,
+    workspace,
+    Position
+} from 'vscode';
+import { LanguageClient } from 'vscode-languageclient/node';
+import { Location as LSLocation } from 'vscode-languageclient';
+
+/**
+ * adopted from https://github.com/microsoft/vscode/blob/5f3e9c120a4407de3e55465588ce788618526eb0/extensions/typescript-language-features/src/languageFeatures/fileReferences.ts
+ */
+export async function addFindFileReferencesListener(
+    getLS: () => LanguageClient,
+    context: ExtensionContext
+) {
+    const disposable = commands.registerCommand('svelte.typescript.findAllFileReferences', handler);
+
+    context.subscriptions.push(disposable);
+
+    async function handler(resource?: Uri) {
+        if (!resource) {
+            resource = window.activeTextEditor?.document.uri;
+        }
+
+        if (!resource || resource.scheme !== 'file') {
+            return;
+        }
+
+        const document = await workspace.openTextDocument(resource);
+
+        await window.withProgress(
+            {
+                location: ProgressLocation.Window,
+                title: 'Finding file references'
+            },
+            async (_, token) => {
+                const locations = await getLS().sendRequest<LSLocation[] | null>(
+                    '$/getFileReferences',
+                    document.uri.toString(),
+                    token
+                );
+
+                const config = workspace.getConfiguration('references');
+                const existingSetting = config.inspect<string>('preferredLocation');
+
+                await config.update('preferredLocation', 'view');
+                try {
+                    await commands.executeCommand(
```

Comment:
@jasonlyu123 This results in VS code showing a little popup of Missing Provider. Ideas?

![image](https://user-images.githubusercontent.com/5598169/170783238-8c02eed0-58e2-4d3e-b882-f83d42242746.png)

Reply:
Ah, right we still need to convert it to vscode's `Location`.

Reply:
Sweet, I got that converted and it's working. 

![image](https://user-images.githubusercontent.com/5598169/170870525-6921cd02-14ff-4e28-8198-3957abfef554.png)